File: Python/Patient.py
# Patient.py
# Kelas turunan Person -> Pasien

from Person import Person

class Patient(Person):
    def __init__(self, nama="", notelp="", alamat="", id_pasien="", penyakit="", status=""):
        super().__init__(nama, notelp, alamat)
        self.__id_pasien = id_pasien
        self.__penyakit = penyakit
        self.__status_rawat = status

    # Setter
    def set_id_pasien(self, id_pasien):
        self.__id_pasien = id_pasien

    def set_penyakit(self, penyakit):
        self.__penyakit = penyakit

    def set_status_rawat(self, status):
        self.__status_rawat = status

    # Getter
    def get_id_pasien(self):
        return self.__id_pasien

    def get_penyakit(self):
        return self.__penyakit

    def get_status_rawat(self):
        return self.__status_rawat

    # Print data pasien
    def print_patient(self):
        print("[Patient] ", end="")
        super().print_info()
        print(f", ID: {self.__id_pasien}, Penyakit: {self.__penyakit}, Status: {self.__status_rawat}")

File: CPP/Patient.cpp
#ifndef PATIENT_CPP
#define PATIENT_CPP

#include <iostream>
#include "Person.cpp"
using namespace std;

class Patient : public Person {
private:
    string idPasien;
    string penyakit;
    string statusRawat;

public:
    Patient() {}
    Patient(string nama, string notelp, string alamat, string id, string penyakit, string status)
        : Person(nama, notelp, alamat) {
        this->idPasien = id;
        this->penyakit = penyakit;
        this->statusRawat = status;
    }

    void setIdPasien(string id) { idPasien = id; }
    void setPenyakit(string penyakit) { this->penyakit = penyakit; }
    void setStatusRawat(string status) { statusRawat = status; }

    string getIdPasien() const { return idPasien; }
    string getPenyakit() const { return penyakit; }
    string getStatusRawat() const { return statusRawat; }

    void printPatient() const {
        cout << "[Patient] "; 
        Person::printInfo();
        cout << ", ID: " << idPasien << ", Penyakit: " << penyakit 
             << ", Status: " << statusRawat << endl;
    }
};

#endif

